distributedlog-core/src/main/java/com/twitter/distributedlog/EnvelopedEntry.java

```diff
@@ -290,7 +290,8 @@ public static InputStream fromInputStream(InputStream src,
         src.reset();
         EnvelopedEntry entry = new EnvelopedEntry(version, statsLogger);
         entry.readFully(new DataInputStream(src));
-        return new ByteArrayInputStream(entry.getDecompressedPayload());
+        return new ByteArrayInputStream(entry.getDecompressedPayload().array());
```

Comment:
i think im missing something here, but it seems like a bad idea to use direct array access since the payload object could have been initialized with a ByteBuffer from anywhere. is this safe?